src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -116,50 +116,81 @@ public void discoverClusterSlots(Jedis jedis) {
     }
   }
 
+  private void discoverClusterSlots(Jedis jedis) {
+    List<Object> slots = jedis.clusterSlots();
+    this.slots.clear();
+
+    for (Object slotInfoObj : slots) {
+      List<Object> slotInfo = (List<Object>) slotInfoObj;
+
+      if (slotInfo.size() <= MASTER_NODE_INDEX) {
+        continue;
+      }
+
+      List<Integer> slotNums = getAssignedSlotArray(slotInfo);
+
+      // hostInfos
+      List<Object> hostInfos = (List<Object>) slotInfo.get(MASTER_NODE_INDEX);
+      if (hostInfos.isEmpty()) {
```

Comment:
It's refactored part of old code "if (hostInfos.isEmpty()) {
 -            continue;" which is not related to PR, I just made it more readable.